server/src/main/java/org/elasticsearch/index/translog/BufferedChecksumStreamInput.java

```diff
@@ -101,7 +101,12 @@ public void reset() throws IOException {
 
     @Override
     public int read() throws IOException {
-        return readByte() & 0xFF;
+        int b = delegate.read();
+        if (b == -1) {
+            return b;
```

Comment:
does very ending value -1 need to be encountered for checksum digest? it used to be in prev. version.
I think prev. version was ok.

Reply:
`-1` is an end of stream marker. It is not a value of the stream.

The checksum was not updated for this value in the previous version since it threw an exception instead of returning `-1`.

Reply:
Thanks for clarification, Tim.